Sources/Puree/Logger.swift

```diff
@@ -19,43 +19,42 @@ public final class Logger {
     }
 
     private let configuration: Configuration
+    private let dispatchQueue = DispatchQueue(label: "com.cookpad.Puree.Logger", qos: .background)
     private(set) var filters: [Filter] = []
     private(set) var outputs: [Output] = []
 
     public var currentDate: Date {
         return configuration.dateProvider.now
     }
 
-    var logStore: LogStore {
-        return configuration.logStore
-    }
-
     public init(configuration: Configuration) throws {
         self.configuration = configuration
 
         try configuration.logStore.prepare()
         try configureFilterPlugins()
         try configureOutputPlugins()
 
-        outputs.forEach { $0.start() }
+        start()
     }
 
     private func configureFilterPlugins() throws {
         filters = try configuration.filterSettings.map { try $0.makeFilter() }
     }
 
     private func configureOutputPlugins() throws {
-        outputs = try configuration.outputSettings.map { try $0.makeOutput(logStore) }
+        outputs = try configuration.outputSettings.map { try $0.makeOutput(configuration.logStore) }
     }
 
     public func postLog(_ payload: [String: Any]?, tag: String) {
-        func matchesOutputs(with tag: String) -> [Output] {
-            return outputs.filter { $0.tagPattern.match(in: tag) != nil }
-        }
+        dispatchQueue.async {
```

Comment:
Could you explain use `async`?

I think there is the case which `output.emit` would be called immediately.

Reply:
This is the main bottleneck function. We needs to reduce this weight.

> there is the case which output.emit would be called immediately.

If we wanna take out `emit` 's result immediately( like a test ), We have to call `suspend` or `shotdown`. Those functions calls `sync`.

But I thought the most important thing is getting `Logger` threadsafe, so this `async` is not so important ...?